File: 3Sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        HashSet<List<Integer>> set = new HashSet<>();
        
        Arrays.sort(nums);
        for(int i = 0; i < nums.length; i++){
            int left = i+1;
            int right  = nums.length-1;
            while(left < right){
            ArrayList<Integer> values = new ArrayList<>();
            if((nums[left] + nums[i] + nums[right]) == 0){
                values.add(nums[i]);
                values.add(nums[left]);
                values.add(nums[right]);
                set.add(values);
                left++;
            }
            if((nums[left] + nums[i] + nums[right]) < 0){
                left++;
            }
            else{
                right--;
            }
        }
    }
    ArrayList<List<Integer>> answer = new ArrayList<>(set);
    return answer;
}
}

File: 3Sum.py
class Solution(object):
    def threeSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        nums.sort()
        arr = []
        for i in range(len(nums)):
            if i > 0 and nums[i] == nums[i-1]:
                continue
            left = i+1
            right = len(nums)-1
            while(left < right):
                threesum = nums[left]+nums[right]+nums[i]
                if threesum < 0:
                    left += 1
                elif threesum > 0:
                    right -= 1
                else:
                    arr.append((nums[i], nums[left], nums[right]))
                    left += 1
                    while(nums[left] == nums[left-1] and left < right):
                        left += 1
        return arr
                
                
